src/NHibernate/Type/AbstractStringType.cs

```diff
@@ -83,5 +122,78 @@ public string ObjectToSQLString(object value, Dialect.Dialect dialect)
 		}
 
 		#endregion
+
+		#region IParameterizedType Members
+
+		public void SetParameterValues(IDictionary<string, string> parameters)
+		{
+			if (parameters == null)
+				return;
+
+			bool? caseSensitive = null;
+			if (parameters.TryGetValue(CaseSensitiveParameterName, out var value))
+			{
+				caseSensitive = !StringComparer.OrdinalIgnoreCase.Equals("false", value);
+			}
+
+			parameters.TryGetValue(ComparerCultureParameterName, out var cultureName);
+			switch (cultureName?.ToLowerInvariant())
+			{
+				case null:
+					if (caseSensitive.HasValue)
+					{
+						Comparer = caseSensitive.Value ? StringComparer.Ordinal : StringComparer.OrdinalIgnoreCase;
+					}
+					// else use default comparer
+					break;
+				case "current":
+					Comparer = caseSensitive == false ? StringComparer.CurrentCultureIgnoreCase : StringComparer.CurrentCulture;
+					break;
+				case "invariant":
+					Comparer = caseSensitive == false ? StringComparer.InvariantCultureIgnoreCase : StringComparer.InvariantCulture;
+					break;
+				case "ordinal":
+					Comparer = caseSensitive == false ? StringComparer.OrdinalIgnoreCase : StringComparer.Ordinal;
+					break;
+				default:
+					var culture = CultureInfo.GetCultureInfo(cultureName);
+					Comparer = StringComparer.Create(culture, caseSensitive == false);
+					break;
+			}
+		}
+
+		#endregion
+
+		#region override of System.Object Members
+
+		/// <summary>
+		/// Determines whether the specified <see cref="Object"/> is equal to this
+		/// <see cref="AbstractStringType"/>.
+		/// </summary>
+		/// <param name="obj">The <see cref="Object"/> to compare with this <c>AbstractStringType</c>.</param>
+		/// <returns><see langword="true"/> if the SqlType, Name and Comparer properties are the same.</returns>
+		public override bool Equals(object obj)
+		{
+			if (!base.Equals(obj))
```

Comment:
Shouldn't it be opposite?

```csharp 
if (base.Equals(obj)) return true;
```

Reply:
Em... but you are saying exactly what I’m asking about. And your code does the opposite.